src/Hydra/Serializer/PartialCollectionViewNormalizer.php

```diff
@@ -164,4 +143,41 @@ private function cursorPaginationFields(array $fields, int $direction, $object)
 
         return $paginationFilters;
     }
+
+    private function populateDataWithCursorBasedPagination(array $data, array $parsed, \Traversable $object, $cursorPaginationAttribute): array
+    {
+        $objects = iterator_to_array($object);
+        $firstObject = current($objects);
+        $lastObject = end($objects);
+
+        $data['hydra:view']['@id'] = IriHelper::createIri($parsed['parts'], $parsed['parameters']);
+
+        if (false !== $lastObject && \is_array($cursorPaginationAttribute)) {
+            $data['hydra:view']['hydra:next'] = IriHelper::createIri($parsed['parts'], array_merge($parsed['parameters'], $this->cursorPaginationFields($cursorPaginationAttribute, 1, $lastObject)));
+        }
+
+        if (false !== $firstObject && \is_array($cursorPaginationAttribute)) {
+            $data['hydra:view']['hydra:previous'] = IriHelper::createIri($parsed['parts'], array_merge($parsed['parameters'], $this->cursorPaginationFields($cursorPaginationAttribute, -1, $firstObject)));
+        }
+
+        return $data;
+    }
+
+    private function populateDataWithPagination(array $data, array $parsed, ?float $currentPage, ?float $lastPage, ?float $itemsPerPage, ?float $pageTotalItems): array
+    {
+        if (null !== $lastPage) {
+            $data['hydra:view']['hydra:first'] = IriHelper::createIri($parsed['parts'], $parsed['parameters'], $this->pageParameterName, 1.);
+            $data['hydra:view']['hydra:last'] = IriHelper::createIri($parsed['parts'], $parsed['parameters'], $this->pageParameterName, $lastPage);
+        }
+
+        if (1. !== $currentPage) {
+            $data['hydra:view']['hydra:previous'] = IriHelper::createIri($parsed['parts'], $parsed['parameters'], $this->pageParameterName, $currentPage - 1.);
+        }
+
+        if ((null !== $lastPage && $currentPage < $lastPage) || (null === $lastPage && null === $lastPage && $pageTotalItems >= $itemsPerPage)) {
+            $data['hydra:view']['hydra:next'] = IriHelper::createIri($parsed['parts'], $parsed['parameters'], $this->pageParameterName, $currentPage + 1.);
+        }
```

Comment:
what about :
```
if ((null !== $lastPage && $currentPage >= $lastPage) || $pageTotalItems < $itemsPerPage) {
    return $data;
}

$data['hydra:view']['hydra:next'] = IriHelper::createIri($parsed['parts'], $parsed['parameters'], $this->pageParameterName, $currentPage + 1.);
```

 anyways `null === $lastPage && null === $lastPage` is there 2 times

Reply:
It's a c/p but I'll take a look.